java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -315,7 +316,10 @@ public void indexingComplete(Set<URL> indexedRoots) {
                 delegates = this.delegates.toArray(new TextDocumentServiceImpl[this.delegates.size()]);
             }
             for (TextDocumentServiceImpl delegate : delegates) {
-                delegate.reRunDiagnostics();
+                ProxyLookup augmentedLookup = new ProxyLookup(Lookups.fixed(delegate.client), Lookup.getDefault());
+                Lookups.executeWith(augmentedLookup, () -> {
```

Comment:
Please note this runs outside of the event loop, this runs when indexing finishes in a background thread, so the client is not set.

Note this also relates to what settings are used during indexing - there are merged settings across all clients for used by the indexing (see `SingleFileOptionsQueryImpl.globalOptions`), but for the editor, we can use the sharper per-client settings.

Reply:
Ah, sorry, I misunderstood this to say this code is not needed. I suspect we don't have a meaningful `OperationContext` (as this runs asynchronously, and we probably should report any progress here anyway). Is there a way to get the `sessionLookup`? I don't think it is strictly necessary, but we could include that, in case it would be useful later.

Reply:
I don't think we have that - but you could eventually expose it from `LspServerState` implemented by `LanguageServerImpl` . Maybe just make a comment that the lookup is different so when an issue pops up, this conversation will be reminded.

Reply:
Thanks - I've added this:
https://github.com/apache/netbeans/pull/6329/commits/40afb5565210f5535e55ca5153d42e19cbb06052